pkg/volume/azure_dd/controllerCommon.go

```diff
@@ -0,0 +1,520 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure_dd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/golang/glog"
+)
+
+type armVmVhdDiskInfo struct {
+	Uri string `json:"uri"`
+}
+type armVmManagedDiskInfo struct {
+	Id string `json:"id"`
+}
+
+type armVmDataDisk struct {
+	Lun          int                   `json:"lun"`
+	Name         string                `json:"name,omitempty"`
+	CreateOption string                `json:"createOption"`
+	ManagedDisk  *armVmManagedDiskInfo `json:"managedDisk,omitempty"`
+	Vhd          *armVmVhdDiskInfo     `json:"vhd,omitempty"`
+	Caching      string                `json:"caching"`
+	DiskSizeGB   int                   `json:"diskSizeGB,omitempty"`
+}
+
+// interfaces used by attacher, deleter to communicate with ARM
+type BlobDiskController interface {
+	CreateDataDisk(dataDiskName string, storageAccountType string, sizeGB int, forceStandAlone bool) (string, error)
+	DeleteDataDisk(diskUri string, wasForced bool) error
+
+	AttachDisk(nodeName string, diskUri string, cacheMode string) (int, error)
+	DetachDisk(nodeName string, hashedDiskUri string) error
+}
+
+type ManagedDiskController interface {
+	CreateDataDisk(diskName string, storageAccountType string, sizeGB int, tags map[string]string) (string, error)
+	DeleteDataDisk(diskUri string) error
+
+	AttachDisk(nodeName string, diskUri string, cacheMode string) (int, error)
+	DetachDisk(nodeName string, hashedDiskUri string) error
+}
+
+var dataDisksPerVM = map[string]int{
```

Comment:
Personally, I would place this in a different file, it takes up a lot of space here that makes the code harder to read.